### ISSUE TITLE
Timeline for migration ADAL to MSAL in AKS

### ISSUE THREAD

**Issue by AllenWen-at-Azure** [Azure/AKS#4576]
Azure Active Directory Authentication Library (ADAL) has been deprecated, all Microsoft support and development for ADAL, including security fixes, ended on June 30, 2023. We are migrating ADAL to Microsoft Authentication Library (MSAL) in AKS images. Listed below are status and ETA for different components (please note that the ETA are subject to change based on release processes of each components).
| Components | Status | ETA |
| -- | -- | -- |
| CSI disk/file/blob node | Done | 2025-01 |
| Cloud node manager | Done | 2025-01 |
| Kubelet credential provider | Done | 2025-01 |
| azure-policy | Done| 2024-11 |
| ama-logs-rs | Done |   |
|Azure Key Vault provider for Secret Store CSI driver | Done| |

**Comment by neeleshizme** [comment]
can someone confirm if this is already picked up by product team and being worked upon?
I see https://github.com/Azure/AKS/issues/4546 this has been auto closed.

**Comment by AllenWen-at-Azure** [comment]
Yes, product team is working on ADAL migration to MSAL for above mentioned components, we are also working on the ETA for other addons/extensions like defender and keda. I will update the list once I have confirmed ETA.

**Comment by neeleshizme** [comment]
awaiting confirmation on fix, commenting to refrain from closure.
@AllenWen-at-Azure requested details shared on https://github.com/Azure/AKS/issues/4546 thanks

**Comment by sjwaight** [comment]
@AllenWen-at-Azure can you please provide an update.
